### PR TITLE
Webpack

### PR DESCRIPTION
**Pull request by cnasikas**
Add webpack support with babel and sass.

The other PRs should be closed.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const path = require('path')
+const webpack = require('webpack')
+const HtmlWebpackPlugin = require('html-webpack-plugin')
+const CleanWebpackPlugin = require('clean-webpack-plugin')
+
+module.exports = {
+  context: path.resolve(__dirname, 'src'),
+  entry: {
+    app: './app.js'
+  },
```
Comment (dionyziz) [inline webpack.config.js:10]: You could write this as `entry: './app.js'`

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  output: {
+    filename: '[name].bundle.js',
+    path: path.resolve(__dirname, 'dist')
+  },
```
Comment (dionyziz) [inline webpack.config.js:14]: Don't we want `devtool: 'source-map'` as well?

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          'css-loader',
+          'sass-loader'
+        ]
+      },
+      {
+        test: /\.js$/,
+        exclude: [/node_modules/],
+        use: [{
+          loader: 'babel-loader',
+          options: { presets: ['es2015'] }
```
Comment (dionyziz) [inline webpack.config.js]: May be better to specify these in a `.babelrc`

File: css/master.css [hunk 4]
```diff
@@ -1,39 +0,0 @@
...
-}
-footer.navbar {
-    position: absolute;
-    bottom: 0;
-    left: 0;
-    right: 0;
-}
-main {
-    padding-bottom: 250px;
-}
```
Comment (dionyziz) [inline css/master.css:39]: Split the following into two separate commits:

1. The conversion of the existing CSS into SASS
2. The introduction of new styling (e.g. the styling of `.support`)

File: src/assets/scss/_navigation.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#header-nav{
+
+  li{
+    position: relative;
+  }
+
+  li:after{
```
Comment (dionyziz) [inline src/assets/scss/_navigation.scss]: Nit. Be consistent in your coding style: 

1. Indent with 4 spaces
2. Leave a space before `{`

File: src/index.html [hunk 6]
```diff
@@ -0,0 +1,91 @@
+<!DOCTYPE html>
```
Comment (dionyziz) [inline src/index.html]: Since you are creating `src/index.html`, don't you also want to delete `index.html` from the project's root directory?

**Review by dionyziz — CHANGES_REQUESTED** [review]

File: webpack.config.js (hunk 1 above)
Comment (cnasikas) [reply webpack.config.js:10]: From the [docs](https://webpack.js.org/configuration/entry-context/#naming):

> If a string or array of strings is passed, the chunk is named main. If an object is passed, each key is the name of a chunk, and the value describes the entrypoint for the chunk.

So in order the output to be the file `app.bundle.js` and not `main.bundle.js` an object has to be passed. That's why the `filename` attribute on `output` is `[name].bundle.js`. `[name]` is the object's key from `entry`

File: webpack.config.js (hunk 2 above)
Comment (cnasikas) [reply webpack.config.js:14]: Yes :) I changed it.

File: src/assets/scss/_navigation.scss (hunk 5 above)
Comment (cnasikas) [reply src/assets/scss/_navigation.scss]: Do you know any good sass/scss linter ? I found this [one](https://atom.io/packages/linter-sass-lint) but you have to create a `.sass-lint.yml` in the project's root folder. Is that ok? Should I commit it?

File: README.md [hunk 7]
```diff
@@ -1 +1,6 @@
 The decrypto.org website.
+
```
Comment (OrfeasLitos) [inline README.md:2]: Please add more detailed setup instructions. `yarn start` is not enough. I tried first installing yarn through my distro's package manager, then `npm install webpack-dev-server` with and without `sudo`, then `npm install webpack` with and without `sudo` and then the error was `Cannot find module 'webpack'` so I gave up.

**Review by OrfeasLitos — COMMENTED** [review]
Make README.md more descriptive

File: README.md (hunk 7 above)
Comment (dionyziz) [reply README.md:2]: You need `yarn install` – we should add that.

Comment (OrfeasLitos) [reply README.md:2]: Full list of commands that made it work:
```
sudo emerge yarn (or install yarn with your package manager)
sudo yarn install
sudo yarn add webpack
sudo yarn add webpack-dev-server
yarn start
```

Comment (OrfeasLitos) [reply README.md:2]: Too many `sudo`s IMO...

File: src/index.html [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+                    <li class='navitem'>
+                        <a class='nav-link' href=''>partners</a>
+                    </li>
+                </ul>
+            </nav>
+            <p>decentralized academic research group
+               for the political applications of cryptography</p>
+            <a href='' class='btn btn-primary'>Meet the team</a>
+        </header>
+        <main class='container'>
```
Comment (OrfeasLitos) [inline src/index.html]: Leave some space between header and main (`<br> </br>` maybe?)

File: README.md (hunk 7 above)
Comment (dionyziz) [reply README.md:2]: You shouldn't need `sudo` to run `yarn install`. You shouldn't need `sudo` to install `yarn` if you're using `npm` (but that's your choice based on your OS). Also `yarn add webpack` and `yarn add webpack-dev-server` should not be necessary. Can you try `yarn install --production=false`? Perhaps it's skipping devDependencies.

Comment (cnasikas) [reply README.md:2]: 1. Install [yarn](https://yarnpkg.com/lang/en/docs/install/#linux-tab)
2. Go inside repo folder i.e `~/workspace/decrypto.org`
3. `yarn install`
4. `yarn start`

This steps should be enough. You shouldn't need `sudo` to run `yarn` commands. Are you sure you have fetch PR #6  and the `package.json` file is there?

We should add more details :)

Comment (OrfeasLitos) [reply README.md:2]: Probably `sudo` was needed because I installed `yarn` with `emerge`. The following worked without any `sudo`:
```
cd into the repo directory
npm install yarn
yarn install
yarn start
```

File: src/index.html (hunk 8 above)
Comment (cnasikas) [reply src/index.html]: Of course. I think is better to leave aesthetic changes for another PR. You have to decide colors, fonts etc before we can proceed. What do you think?

Comment (OrfeasLitos) [reply src/index.html]: LGTM

File: README.md (hunk 7 above)
Comment (OrfeasLitos) [reply README.md:2]: This way is also cleaner because if I do `rm -rf decrypto.org` there is no trace of anything left on my machine.

File: src/index.html (hunk 6 above)
Comment (cnasikas) [reply src/index.html]: Yes, I forgot it.

File: src/assets/scss/_navigation.scss (hunk 5 above)
Comment (dionyziz) [reply src/assets/scss/_navigation.scss]: Yes, please add `.sass-lint.yml`. Of course we'll want to use the `sass-lint` package, not the respective atom package.

File: README.md (hunk 7 above)
Comment (cnasikas) [reply README.md:2]: Done. I pass the `npm` step because of:

>Note: Installation of Yarn via npm is generally not recommended. When installing Yarn with Node-based package managers, the package is not signed, and the only integrity check performed is a basic SHA1 hash, which is a security risk when installing system-wide apps.

>For these reasons, it is highly recommended that you install Yarn through the installation method best suited to your operating system.

Also `npm install yarn` installs yarn locally, per project. It is better to install it globally (`npm install --global yarn`) if you are using it to other projects

Comment (OrfeasLitos) [reply README.md:2]: But then a bunch of `sudo`s are needed, some of them also installing other things.

Comment (OrfeasLitos) [reply README.md:2]: I'm not sure which practice is better as a matter of fact...

**Comment by cnasikas** [comment]
All the changes have been made!

**Comment by dionyziz** [comment]
Consider adhering to this warning:

```
warning babel-preset-es2015@6.24.1: We're super 😸  excited that you're trying to use ES2015 syntax, but instead of making more yearly presets 😭 , Babel now has a better preset that we recommend you use instead: npm install babel-preset-env --save-dev
```

Let's switch to babel-preset-env?

File: src/index.html [hunk 9]
```diff
@@ -4,13 +4,11 @@
         <meta charset='utf-8'>
         <meta name='viewport' content='width=device-width, initial-scale=1, shrink-to-fit=no'>
         <title>de·crypto</title>
-        <link rel='stylesheet' href='node_modules/bootstrap/dist/css/bootstrap.min.css' />
-        <link rel="stylesheet" href="css/master.css">
     </head>
     <body>
-        <header class='text-center'>
+        <header id="header" class='text-center'>
```
Comment (dionyziz) [inline src/index.html]: Be consistent in your quotation marks. Use `'` everywhere in this file.

**Comment by dionyziz** [comment]
When making stylistic changes, please provide a screenshot for the review.

**Review by dionyziz — APPROVED** [review]
LGTM. Please fix the remaining issues and merge.

**Comment by cnasikas** [comment]
> Consider adhering to this warning:
>
> ```warning babel-preset-es2015@6.24.1: We're super 😸  excited that you're trying to use ES2015 > syntax, but instead of making more yearly presets 😭 , Babel now has a better preset that we recommend you use instead: npm install babel-preset-env --save-dev```
Let's switch to babel-preset-env?

I don't get this warning. Strange, I should.
